{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "daf53cf9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>p_value</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>ci_lower</th>\n",
       "      <th>ci_upper</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs61769339</td>\n",
       "      <td>0.532266</td>\n",
       "      <td>1</td>\n",
       "      <td>662622</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.110178</td>\n",
       "      <td>0.012680</td>\n",
       "      <td>0.020303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs190214723</td>\n",
       "      <td>0.870407</td>\n",
       "      <td>1</td>\n",
       "      <td>693625</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.950775</td>\n",
       "      <td>-0.006036</td>\n",
       "      <td>0.037000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id   p_value  chromosome  base_pair_location effect_allele  \\\n",
       "0   rs61769339  0.532266           1              662622             A   \n",
       "1  rs190214723  0.870407           1              693625             T   \n",
       "\n",
       "  other_allele  effect_allele_frequency      beta  standard_error  odds_ratio  \\\n",
       "0            G                 0.110178  0.012680        0.020303         NaN   \n",
       "1            C                 0.950775 -0.006036        0.037000         NaN   \n",
       "\n",
       "   ci_lower  ci_upper  \n",
       "0       NaN       NaN  \n",
       "1       NaN       NaN  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# inspecció arxius GWAS sense harmonitzar (majoria GRCh37)\n",
    "\n",
    "import pandas as pd\n",
    "gwas = pd.read_csv(r\"D:\\TFM\\GWAS originals i harmonitzats\\alzheimer_33589840-GCST90012877-EFO_0000249-Build37.f.tsv.gz\", sep='\\t')\n",
    "gwas.head(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "784960ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>panel_variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>non_effect_allele</th>\n",
       "      <th>frequency</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>zscore</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>sample_size</th>\n",
       "      <th>n_cases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs61769339</td>\n",
       "      <td>chr1_727242_G_A_b38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>727242</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.110178</td>\n",
       "      <td>0.532266</td>\n",
       "      <td>0.624550</td>\n",
       "      <td>0.012680</td>\n",
       "      <td>0.020303</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs12238997</td>\n",
       "      <td>chr1_758351_A_G_b38</td>\n",
       "      <td>chr1</td>\n",
       "      <td>758351</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.115854</td>\n",
       "      <td>0.834508</td>\n",
       "      <td>-0.208924</td>\n",
       "      <td>-0.003903</td>\n",
       "      <td>0.018679</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_id     panel_variant_id chromosome  position effect_allele  \\\n",
       "0  rs61769339  chr1_727242_G_A_b38       chr1    727242             A   \n",
       "1  rs12238997  chr1_758351_A_G_b38       chr1    758351             G   \n",
       "\n",
       "  non_effect_allele  frequency    pvalue    zscore  effect_size  \\\n",
       "0                 G   0.110178  0.532266  0.624550     0.012680   \n",
       "1                 A   0.115854  0.834508 -0.208924    -0.003903   \n",
       "\n",
       "   standard_error  sample_size  n_cases  \n",
       "0        0.020303          NaN      NaN  \n",
       "1        0.018679          NaN      NaN  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# inspecció arxius GWAS harmonitzats a GRCh38\n",
    "\n",
    "import pandas as pd\n",
    "gwas_harm = pd.read_csv(r\"D:\\TFM\\GWAS originals i harmonitzats\\alzheimer_harm.txt.gz\", sep='\\t')\n",
    "gwas_harm.head(2)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3948554c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "weights mashr:\n",
      " [('ENSG00000169583.12', 'rs908836', 'chr9_136996001_G_A_b38', 'G', 'A', -0.18060518974537)]\n",
      "extra mashr:\n",
      " [('ENSG00000000457.13', 'SCYL3', 'protein_coding', 2, None, None, None)]\n"
     ]
    }
   ],
   "source": [
    "# inspecció arxius models de predicció MASHR\n",
    "# 2 taules: weights, extra\n",
    "\n",
    "import pandas as pd\n",
    "import sqlite3\n",
    "\n",
    "con = sqlite3.connect(\"D:\\TFM\\models\\mashr\\mashr_Brain_Amygdala.db\")\n",
    "cur = con.cursor()\n",
    "cur.execute(\"select * from weights limit 1;\")\n",
    "weights = cur.fetchall()\n",
    "print(\"weights mashr:\\n\", weights)\n",
    "cur.execute(\"select * from extra limit 1;\")\n",
    "extra = cur.fetchall()\n",
    "print(\"extra mashr:\\n\",extra)\n",
    "con.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2216dccd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "weights elastic net:\n",
      " [('ENSG00000185736.15', 'rs10903832', 'chr10_275014_A_G_b38', 'A', 'G', -0.003623735942687525)]\n",
      "extra elastic net:\n",
      " [('ENSG00000185736.15', 'ADARB2', 'protein_coding', 0.5, 3230, 12, 0.007408914871938244, 0.07623175174537218, 0.03307520053347509, 0.05699649246030865, 0.13514629160320268, 0.05583746495800605, 5, 0.2800346356299044, 0.11554214811599335, 2.9553417565426283, 0.07841939715237331, 0.0031232287052839845, None)]\n"
     ]
    }
   ],
   "source": [
    "# inspecció arxius models de predicció Elastic Net\n",
    "# 2 taules: weights, extra\n",
    "\n",
    "import pandas as pd\n",
    "import sqlite3\n",
    "\n",
    "con = sqlite3.connect(\"D:\\TFM\\models\\elasticnet\\en_Brain_Amygdala.db\")\n",
    "cur = con.cursor()\n",
    "cur.execute(\"select * from weights limit 1;\")\n",
    "weights = cur.fetchall()\n",
    "print(\"weights elastic net:\\n\", weights)\n",
    "cur.execute(\"select * from extra limit 1;\")\n",
    "extra = cur.fetchall()\n",
    "print(\"extra elastic net:\\n\", extra)\n",
    "con.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7405f16d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "weights psychendode:\n",
      " [('ENSG00000000457', 'rs6703487', 'chr1_169550749_C_T_b37', 'C', 'T', 0.0010487355157559)]\n",
      "extra psychencode :\n",
      " [('ENSG00000000457', None, 26, None, None, None)]\n",
      "weights psychendode2:\n",
      " [('ENSG00000000457', 'rs6703487', 'chr1_169550749_C_T_b37', 'C', 'T', 0.0010487355157559)]\n",
      "extra psychencode2 :\n",
      " [('ENSG00000000457', 'SCYL3', 26.0, 0.0297869829429901, 1.62680279346564e-10, None)]\n"
     ]
    }
   ],
   "source": [
    "# inspecció arxius models de predicció Psychencode\n",
    "# 2 taules: weights, extra\n",
    "\n",
    "import pandas as pd\n",
    "import sqlite3\n",
    "\n",
    "con = sqlite3.connect(\"D:\\TFM\\models\\psychencode\\psychencode.db\")\n",
    "cur = con.cursor()\n",
    "cur.execute(\"select * from weights limit 1;\")\n",
    "weights = cur.fetchall()\n",
    "print(\"weights psychendode:\\n\", weights)\n",
    "cur.execute(\"select * from extra limit 1;\")\n",
    "extra = cur.fetchall()\n",
    "print(\"extra psychencode :\\n\", extra)\n",
    "con.close()\n",
    "\n",
    "con = sqlite3.connect(\"D:\\TFM\\models\\psychencode\\psychencode_hg19_annotR2.db\")\n",
    "cur = con.cursor()\n",
    "cur.execute(\"select * from weights limit 1;\")\n",
    "weights = cur.fetchall()\n",
    "print(\"weights psychendode2:\\n\", weights)\n",
    "cur.execute(\"select * from extra limit 1;\")\n",
    "extra = cur.fetchall()\n",
    "print(\"extra psychencode2 :\\n\", extra)\n",
    "con.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0a047490",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE RSID1 RSID2 VALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000457.13 chr1_169894024_A_C_b38 chr1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000457.13 chr1_169894024_A_C_b38 chr1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              GENE RSID1 RSID2 VALUE\n",
       "0  ENSG00000000457.13 chr1_169894024_A_C_b38 chr1...\n",
       "1  ENSG00000000457.13 chr1_169894024_A_C_b38 chr1..."
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# inspecció arxius matriu de covariància MASHR\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "cov_mashr = pd.read_csv(\"D:\\TFM\\models\\mashr\\mashr_Brain_Amygdala.txt.gz\")\n",
    "cov_mashr.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fc1ce54f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE RSID1 RSID2 VALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000185736.15 rs10903832 rs10903832 0.204...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000185736.15 rs10903832 rs7100066 -0.040...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              GENE RSID1 RSID2 VALUE\n",
       "0  ENSG00000185736.15 rs10903832 rs10903832 0.204...\n",
       "1  ENSG00000185736.15 rs10903832 rs7100066 -0.040..."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# inspecció arxius matriu de covariància Elastic Net\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "cov_elastic = pd.read_csv(\"D:\\TFM\\models\\elasticnet\\en_Brain_Amygdala.txt.gz\", sep='\\t')\n",
    "display(cov_elastic.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "732ba203",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE RSID1 RSID2 VALUE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000003249 rs140580708 rs140580708 0.0130...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000003249 rs140580708 rs164739 -0.008913...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              GENE RSID1 RSID2 VALUE\n",
       "0  ENSG00000003249 rs140580708 rs140580708 0.0130...\n",
       "1  ENSG00000003249 rs140580708 rs164739 -0.008913..."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# inspecció arxius matriu de covariància Psychencode\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "cov_psychencode = pd.read_csv(\"D:\\TFM\\models\\psychencode\\psychencode.txt.gz\", sep='\\t')\n",
    "display(cov_psychencode.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ffdd7faf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>zscore</th>\n",
       "      <th>effect_size</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>var_g</th>\n",
       "      <th>pred_perf_r2</th>\n",
       "      <th>pred_perf_pval</th>\n",
       "      <th>pred_perf_qval</th>\n",
       "      <th>n_snps_used</th>\n",
       "      <th>n_snps_in_cov</th>\n",
       "      <th>n_snps_in_model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000130203.9</td>\n",
       "      <td>APOE</td>\n",
       "      <td>29.611221</td>\n",
       "      <td>9.781687</td>\n",
       "      <td>1.071481e-192</td>\n",
       "      <td>3.576995e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000267467.3</td>\n",
       "      <td>APOC4</td>\n",
       "      <td>13.742450</td>\n",
       "      <td>122357.623798</td>\n",
       "      <td>5.653013e-43</td>\n",
       "      <td>9.114232e-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000186567.12</td>\n",
       "      <td>CEACAM19</td>\n",
       "      <td>10.992327</td>\n",
       "      <td>0.279253</td>\n",
       "      <td>4.160585e-28</td>\n",
       "      <td>5.189932e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000104853.15</td>\n",
       "      <td>CLPTM1</td>\n",
       "      <td>10.663039</td>\n",
       "      <td>7.920193</td>\n",
       "      <td>1.515621e-26</td>\n",
       "      <td>1.039973e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000203710.10</td>\n",
       "      <td>CR1</td>\n",
       "      <td>8.751949</td>\n",
       "      <td>0.263279</td>\n",
       "      <td>2.096986e-18</td>\n",
       "      <td>4.296938e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 gene gene_name     zscore    effect_size         pvalue  \\\n",
       "0   ENSG00000130203.9      APOE  29.611221       9.781687  1.071481e-192   \n",
       "1   ENSG00000267467.3     APOC4  13.742450  122357.623798   5.653013e-43   \n",
       "2  ENSG00000186567.12  CEACAM19  10.992327       0.279253   4.160585e-28   \n",
       "3  ENSG00000104853.15    CLPTM1  10.663039       7.920193   1.515621e-26   \n",
       "4  ENSG00000203710.10       CR1   8.751949       0.263279   2.096986e-18   \n",
       "\n",
       "          var_g  pred_perf_r2  pred_perf_pval  pred_perf_qval  n_snps_used  \\\n",
       "0  3.576995e-04           NaN             NaN             NaN            1   \n",
       "1  9.114232e-13           NaN             NaN             NaN            1   \n",
       "2  5.189932e-02           NaN             NaN             NaN            2   \n",
       "3  1.039973e-04           NaN             NaN             NaN            1   \n",
       "4  4.296938e-02           NaN             NaN             NaN            1   \n",
       "\n",
       "   n_snps_in_cov  n_snps_in_model  \n",
       "0              1                1  \n",
       "1              1                1  \n",
       "2              2                2  \n",
       "3              1                1  \n",
       "4              1                1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# inspecció arxius generats per SPrediXcan\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "exemple = pd.read_csv(r\"D:\\TFM\\SPrediXcan\\alzheimer\\alzheimer_amygdala.csv\",sep =\",\")\n",
    "exemple.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
